PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -431,11 +434,165 @@ UpdateContainerComp (
   // Component base = Container base + data offset from container base + offset of component inside container
   //
   ComponentBase = ContainerEntryPtr->Base + ContainerHdr->DataOffset + ComponentEntryPtr->Offset;
+
+  // Check Svn for container component
+  FlashCompLzHeader = (LOADER_COMPRESSED_HEADER *) (UINTN) ComponentBase;
+  CapCompLzHeader   = (LOADER_COMPRESSED_HEADER *) ((UINTN)ImageHdr + sizeof(EFI_FW_MGMT_CAP_IMAGE_HEADER));
+  if (CapCompLzHeader->Svn < FlashCompLzHeader->Svn) {
+    DEBUG((DEBUG_INFO, "Container Component svn did not met!"));
+    return EFI_UNSUPPORTED;
+  }
+
   Status = UpdateSingleComponent (ComponentBase, ComponentEntryPtr->Size, ImageHdr);
 
   return Status;
 }
 
+
+/**
+  Locate components in a container based on component name
+
+  This function will search for components in a container based
+  on name of component provided.
+
+  @param[in]      ContainerHdr       Pointer to container header
+  @param[in]      ComponentName      Name of the component to retrieve
+  @param[in/out]  ComponentAddr      Componet header address
+
+
+  @retval  EFI_SUCCESS      Component found successful
+  @retval  other            error occurred during firmware update
+**/
+EFI_STATUS
+GetComponentAddr (

```

Comment:
why do we need a new function ? The following will give you the base of the component in flash.
LocateComponent (ContainerSig,  Component,  &Base, &Size);

Reply:
This is a generic function to search component and return its comp address. It takes containerhdr as input.
This is required to retrieve components in **Capsule** container image.
As part of SVN verification, we find a component in flash and next find a component in capsule container. Later check SVN's,

Reply:
Using "LocateComponentEntryFromContainer" instead of new function